packages/prepo-shared-contracts/test/SafeAccessControlEnumerable.test.ts

```diff
@@ -0,0 +1,211 @@
+    })
+
+    it('emits RoleAdminNomineeUpdate', async () => {
+      const previousNominee = await safeAccessControlEnumerable.getRoleAdmin(roleB)
+
+      const tx = await safeAccessControlEnumerable.connect(owner).setRoleAdminNominee(roleB, roleA)
+
+      await expect(tx)
+        .to.emit(safeAccessControlEnumerable, 'RoleAdminNomineeUpdate')
+        .withArgs(previousNominee, roleA)
+    })
+  })
+
+  describe('# acceptRoleAdmin', () => {
+    beforeEach(async () => {
+      roleAMember = user1
+      await setupSafeAccessControlEnumerable()
+      await safeAccessControlEnumerable.connect(owner).grantRole(roleA, roleAMember.address)
+      await safeAccessControlEnumerable.connect(roleAMember).acceptRole(roleA)
+      expect(await safeAccessControlEnumerable.hasRole(roleA, roleAMember.address)).to.eq(true)
+      await safeAccessControlEnumerable.connect(owner).setRoleAdminNominee(roleB, roleA)
+    })
+
+    it('reverts if not role admin and not role admin nominee', async () => {
+      await safeAccessControlEnumerable.connect(owner).setRoleAdminNominee(roleB, ZERO_HASH)
+      const roleBAdmin = await safeAccessControlEnumerable.getRoleAdmin(roleB)
+      const roleBAdminNominee = await safeAccessControlEnumerable.getRoleAdminNominee(roleB)
+      expect(await safeAccessControlEnumerable.hasRole(roleBAdmin, user1.address)).to.eq(false)
+      expect(await safeAccessControlEnumerable.hasRole(roleBAdminNominee, user1.address)).to.eq(
+        false
+      )
+
+      await expect(
+        safeAccessControlEnumerable.connect(user1).acceptRoleAdmin(roleB)
+      ).to.be.revertedWith(
+        `AccessControl: account ${user1.address.toLowerCase()} is missing role ${roleBAdminNominee}`
+      )
+    })
+
+    it('reverts if role admin but not role admin nominee', async () => {
+      const roleBAdmin = await safeAccessControlEnumerable.getRoleAdmin(roleB)
+      const roleBAdminNominee = await safeAccessControlEnumerable.getRoleAdminNominee(roleB)
+      expect(await safeAccessControlEnumerable.hasRole(roleBAdmin, owner.address)).to.eq(true)
+      expect(await safeAccessControlEnumerable.hasRole(roleBAdminNominee, owner.address)).to.eq(
+        false
+      )
+
+      await expect(
+        safeAccessControlEnumerable.connect(owner).acceptRoleAdmin(roleB)
+      ).to.be.revertedWith(
+        `AccessControl: account ${owner.address.toLowerCase()} is missing role ${roleBAdminNominee}`
+      )
+    })
+
+    it('sets role admin to role admin nominee', async () => {
+      const roleBAdminNominee = await safeAccessControlEnumerable.getRoleAdminNominee(roleB)
+      expect(await safeAccessControlEnumerable.getRoleAdmin(roleB)).to.not.eq(roleBAdminNominee)
+
+      await safeAccessControlEnumerable.connect(roleAMember).acceptRoleAdmin(roleB)
+
+      expect(await safeAccessControlEnumerable.getRoleAdmin(roleB)).to.eq(roleBAdminNominee)
+    })
+
+    it('sets role admin nominee to zero', async () => {
```

Comment:
```suggestion
    it('sets role admin nominee to default admin afterwards', async () => {
```
I think we need to add some detail here because this could be confused with our other testcase nomenclature when we are explicitly setting a value to zero.

Lets also use `DEFAULT_ADMIN_ROLE` throughout this test suite rather than `ZERO_HASH`.